frame/staking/src/lib.rs

```diff
@@ -2553,21 +2624,46 @@ impl<T: Trait> Module<T> {
 		compute: ElectionCompute,
 		claimed_score: PhragmenScore,
 		era: EraIndex,
-	) -> Result<(), Error<T>> {
+		election_size: ElectionSize,
+	) -> DispatchResultWithPostInfo {
 		// Do the basic checks. era, claimed score and window open.
 		Self::pre_dispatch_checks(claimed_score, era)?;
+		// the weight that we will refund in case of a correct submission. We compute this now
+		// because the data needed for it will be consumed further down.
+		let adjusted_weight = weight::weight_for_correct_submit_solution::<T>(
+			&winners,
+			&compact_assignments,
+			&election_size,
+		);
 
 		// Check that the number of presented winners is sane. Most often we have more candidates
-		// that we need. Then it should be Self::validator_count(). Else it should be all the
+		// than we need. Then it should be `Self::validator_count()`. Else it should be all the
 		// candidates.
-		let snapshot_length = <SnapshotValidators<T>>::decode_len()
+		let snapshot_validators_length = <SnapshotValidators<T>>::decode_len()
+			.map(|l| l as u32)
 			.ok_or_else(|| Error::<T>::SnapshotUnavailable)?;
 
+		// size of the solution must be correct.
+		ensure!(
+			snapshot_validators_length == election_size.validators.into(),
```

Comment:
```suggestion
			snapshot_validators_length == u32::from(election_size.validators),
```
This fixed compilation for me somehow 🤷

I just explicited as much as possible, then rustc said another line was unable to do type inference, and then this line, 🤔